modules/openPairIdSystem.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * This module adds PAIR Id to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/pairIdSystem
+ * @requires module:modules/userId
+ */
+
+import { submodule } from '../src/hook.js';
+import {getStorageManager} from '../src/storageManager.js'
+import { logInfo } from '../src/utils.js';
+import {MODULE_TYPE_UID} from '../src/activities/modules.js';
+import {VENDORLESS_GVLID} from '../src/consentHandler.js';
+
+/**
+ * @typedef {import('../modules/userId/index.js').Submodule} Submodule
+ */
+
+const MODULE_NAME = 'openPairId';
+const PAIR_ID_KEY = 'pairId';
+
+const DEFAULT_STORAGE_PAID_ID_KEYS = {
+  liveramp: '_lr_pairId'
+};
+
+export const storage = getStorageManager({moduleType: MODULE_TYPE_UID, moduleName: MODULE_NAME});
+
+function pairIdFromLocalStorage(key) {
+  return storage.localStorageIsEnabled() ? storage.getDataFromLocalStorage(key) : null;
+}
+
+function pairIdFromCookie(key) {
+  return storage.cookiesAreEnabled() ? storage.getCookie(key) : null;
+}
+
+/** @type {Submodule} */
+export const pairIdSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+  /**
+   * used to specify vendor id
+   * @type {number}
+   */
+  gvlid: VENDORLESS_GVLID,
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @param { string | undefined } value
+   * @returns {{pairId:string} | undefined }
+   */
+  decode(value) {
+    return value && Array.isArray(value) ? {'pairId': value} : undefined
```

Comment:
This should be `openPairId`  - if it's meant to be distinct from the other module - or if they're meant to be mutually exclusive (clashes are not a problem),  the EID configuration in this should also use `pairId`.